core/vm/contracts.go

```diff
@@ -528,7 +529,7 @@ func RedeemQuai(blockContext BlockContext, statedb StateDB, sender common.Addres
 		return gasUsed, stateGas, errors.New("account has locked too many times, overflows uint64")
 	}
 
-	for i := int64(0); i < math.MaxInt64; i++ { // TODO: We should decide on a reasonable limit
```

Comment:
is MaxInt64 a reasonable limit?

Reply:
Yes because the caller will run out of gas well before that limit is hit